074-additional-volumes-support.md

```diff
@@ -0,0 +1,127 @@
+|CruiseControl    |Kafka            |spec -> cruiseControl -> template -> pod/cruiseControlContainer                       |CruiseControl.java           |
+|KafkaExporter    |Kafka            |spec -> kafkaExporter -> template -> pod/container                                    |KafkaExporter.java           |
+|KafkaConnect     |KafkaConnect     |spec -> template -> pod/connectContainer/initContainer/buildContainer                 |KafkaConnectCluster.java     |
+|KafkaBridge      |KafkaBridge      |spec -> template -> pod/bridgeContainer/initContainer                                 |KafkaBridgeCluster.java      |
+|KafkaMirrorMaker2|KafkaMirrorMaker |spec -> template -> pod/connectContainer/initContainer/buildContainer                 |KafkaMirrorMaker2Cluster.java|
+
+
+It is proposed to not add support in the following locations - if `volumes` or `volumeMounts` are specified in the PodTemplate or ContainerTemplate in any of these locations it will be ignored:
+
+|Pod             |CRD             |Schema Location                                          |Reason                                 |
+|----------------|----------------|---------------------------------------------------------|---------------------------------------|
+|KafkaMirrorMaker|KafkaMirrorMaker|spec -> template -> pod/mirrorMakerContainer             |KafkaMirrorMaker has been deprecated   |
+|Kafka           |Kafka           |spec -> jmxTrans -> template -> pod/container            |JmxTrans no longer supported           |
+|EntityOperator  |Kafka           |spec -> entityOperator -> template -> tlsSidecarContainer|tlsSidecarContainer no longer supported|
+|CruiseControl   |Kafka           |spec -> cruiseControl -> template -> tlsSidecarContainer |tlsSidecarContainer deprecated         |
+
+
+An example configuration could look as follows:
+
+```yaml
+    kind: Kafka
+    spec:
+      kafka:
+        template:
+          pod:
+            volumes:
+            - name: example-configmap
+              configMap:
+                name: config-map-name
+            - name: temp
+              emptyDir: {}
+            - name: example-csi-volume
+              csi:
+                driver: csi-driver-name
+                readOnly: true
+                volumeAttributes:
+                  secretProviderClass: example-secret-provider-class
+            - name: example-projected-volume
+              projected:
+                sources:
+                  serviceAccountToken
+          kafkaContainer:
+            volumeMounts:
+            - name: example-configmap
+              path: /path/to/mount/cm-volume
+            - name: temp
+              path: /tmp/logs
+            - name: example-csi-volume
+              path: /path/to/mount/csi-volume
+            - name: example-projected-volume
+              path: /path/to/mount/proj-volume
+
+```
+
+### Supported volumes

```

Comment:
Some stuff I'm wondering about ...
* Should [PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#claims-as-volumes) be added to the list as well? Wouldn't PVCs be the most common way to create the volume?
* Should Projected volumes be really on the list? What do you need it for? Do you have any use-case? It does not seem to match any of the motivations for this I'm aware of. (It does not seem to be part of the PR anyway)

Reply:
We will rework this. We will remove projected volumes and look more into CSI volumes vs. PVC

Reply:
The CSI volumes actually might have had some value for the CSI secrets. But its fine, we can limit the scope and add it later when needed 👍